modules/artifact/src/main/java/org/mule/module/artifact/classloader/DefaultResourceReleaser.java

```diff
@@ -187,24 +194,95 @@ private boolean hasDeclaredMethod(Class<?> clazz, String methodName, Class<?>...
    * Workaround for http://bugs.mysql.com/bug.php?id=65909
    */
   private void shutdownMySqlAbandonedConnectionCleanupThread() {
-
-    Class<?> classAbandonedConnectionCleanupThread;
     try {
-      classAbandonedConnectionCleanupThread =
-          this.getClass().getClassLoader().loadClass("com.mysql.jdbc.AbandonedConnectionCleanupThread");
+      Class<?> cleanupThreadsClass = findMySqlDriverClass();
+      if (cleanupThreadsClass == null) {
```

Comment:
remove this block. `findMySqlDriverClass` should either fail with a ClassNotFound, or return an optional after logging that no class was found. We then do the rest of the logic as part of the `ifPresent`